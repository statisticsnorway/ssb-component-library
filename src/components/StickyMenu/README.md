StickyMenu
========

> A page layout component to wrap around content 

### Usage

#### HTML

We only provide the container making the header sticky.
The content is just a suggestion.
```html
<div class="ssb-sticky-menu">
    <div class="menu-content">
    <!-- Suggested content -->
        <div style="display: flex; justify-content: space-between; width: 100%;">
            <div>
                <span class="ssb-text-wrapper small-text">Kommunefakta</span>
                <div style="font-size: 24px; font-weight: bold;">Stordal (Møre og Romsdal)</div>
            </div>
            <div style="align-items: center; display: flex;">
                <div style="align-items: center; display: flex;">
                    <label for="kommune" class="input-header">Bytt kommune:</label>
                    <div class="ssb-input">
                        <div class="input-wrapper">
                            <input id="kommune" type="text" placeholder="Søk på kommune" aria-label="Søk på kommune" class=" with-icon" value="" />
                            <button class="icon-wrapper search-icon"><i>{feather.search 18px}</i></button>
                        </div>
                    </div>
                </div>
                <div class="choose-chart" style="margin-left: 30px;">Velg i kart</div>
                {feather.chevronDown 18px}
            </div>
        </div>
    </div>
</div>
```

#### React

```jsx harmony
<StickyMenu>
    Insert content here
</StickyMenu>
```

Available props:

| Name       | Type           | Description  |
| ---------- | ------------- | ----- |
| children   | Required node | Content within the wrapper |
