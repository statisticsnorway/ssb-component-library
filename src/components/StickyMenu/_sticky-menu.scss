.ssb-sticky-menu {
  @include roboto;
  align-items: center;
  background-color: $ssb-white;
  border: solid 1px $ssb-dark-2;
  box-shadow: 2px 2px 10px 0 rgba(206, 205, 205, .5);
  color: $ssb-dark-6;
  display: flex;
  justify-content: center;
  line-height: normal;
  padding-bottom: 25px;
  padding-top: 25px;
  position: sticky;
  top: 0;
  width: 100%;

  .menu-content {
    max-width: 1200px;
    width: 100%;
  }

  .chevron-down-icon {
    color: $ssb-green-4;
    font-size: 18px;
    margin-right: 2px;
  }

  input {
	min-width: 180px;
	width: 100%;
  }

  .input-header {
    color: $ssb-dark-5;
    font-size: 18px;
	font-weight: bold;
	margin-right: 10px;
	white-space: nowrap;
  }

  .title{
    font-size: 24px;
    font-weight: bold;
  }

}
