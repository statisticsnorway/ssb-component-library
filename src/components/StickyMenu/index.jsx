import React from 'react';
import PropTypes from 'prop-types';

const StickyMenu = ({
	children,
}) => (
	<div className="ssb-sticky-menu">
		<div className="menu-content">{children}</div>
	</div>
);

StickyMenu.defaultProps = {};

StickyMenu.propTypes = {
	children: PropTypes.node.isRequired,
};

export default StickyMenu;
