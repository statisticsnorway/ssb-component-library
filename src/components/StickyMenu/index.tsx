import React, { type ReactNode } from 'react'
import classNames from '../../utils/utils'

interface StickyMenuProps {
  children: ReactNode
  className?: string
}

const StickyMenu: React.FC<StickyMenuProps> = ({ children, className }) => (
  <div className={classNames('ssb-sticky-menu', className)}>
    <div className='menu-content'>{children}</div>
  </div>
)

export default StickyMenu
