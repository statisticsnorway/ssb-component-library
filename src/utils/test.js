import fs from 'fs';
import path from 'path';
import { render as rtlRender } from '@testing-library/react';

export * from '@testing-library/react';
export * from '@testing-library/jest-dom';

const styles = fs.readFileSync(path.relative(process.cwd(), './lib/bundle.css'), 'utf8');

export function render(ui, options) {
	const view = rtlRender(ui, options);

	const styleElement = document.createElement('style');
	styleElement.innerHTML = styles;
	document.body.appendChild(styleElement);

	return view;
}
