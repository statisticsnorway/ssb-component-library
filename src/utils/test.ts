import fs from 'fs'
import path from 'path'
import { RenderOptions, RenderResult, render as rtlRender } from '@testing-library/react'
import { ReactNode } from 'react'

export * from '@testing-library/react'

const styles = fs.readFileSync(path.relative(process.cwd(), './lib/bundle.css'), 'utf8')

export function render(ui: ReactNode, options?: RenderOptions): RenderResult {
  const view = rtlRender(ui, options)

  const styleElement = document.createElement('style')
  styleElement.innerHTML = styles
  document.body.appendChild(styleElement)

  return view
}
