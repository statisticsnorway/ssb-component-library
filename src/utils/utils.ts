// utils.ts
const classNames = (...classes: (string | boolean | undefined)[]): string => classes.filter(Boolean).join(' ')

export default classNames
