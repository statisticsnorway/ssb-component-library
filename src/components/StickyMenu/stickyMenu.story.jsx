import React from 'react';
import { storiesOf } from '@storybook/react';
import centered from '@storybook/addon-centered/react';
import { ChevronDown } from 'react-feather';
import StickyMenu from './index';
import Input from '../Input';
import Text from '../Text';

storiesOf('StickyMenu', module).addDecorator(centered)
	.add('Default', () => (
		<div style={{ width: '1400px' }}>
			<StickyMenu>
				<div style={{ display: 'flex', justifyContent: 'space-between', width: '100%' }}>
					<div>
						<Text small>Kommunefakta</Text>
						<div style={{ fontSize: '24px', fontWeight: 'bold' }}>Stordal (Møre og Romsdal)</div>
					</div>
					<div style={{ alignItems: 'center', display: 'flex' }}>
						<div style={{ alignItems: 'center', display: 'flex' }}>
							<label htmlFor="kommune" className="input-header">Bytt kommune:</label>{/* eslint-disable-line */}
							<Input id="kommune" ariaLabel="Søk på kommune" searchField placeholder="Søk på kommune" />
						</div>
						<div style={{ marginLeft: '30px' }} className="choose-chart">Velg i kart</div>
						<ChevronDown size={18} className="chevron-down-icon" />
					</div>
				</div>
			</StickyMenu>
		</div>
	));
