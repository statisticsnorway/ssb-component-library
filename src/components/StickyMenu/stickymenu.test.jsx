import React from 'react'
import { render } from '../../utils/test'
import StickyMenu from './index'

describe('StickyMenu component', () => {
  test('Matches the snapshot', () => {
    const { asFragment } = render(<StickyMenu>StickyMenu</StickyMenu>)
    expect(asFragment()).toMatchSnapshot()
  })
})
