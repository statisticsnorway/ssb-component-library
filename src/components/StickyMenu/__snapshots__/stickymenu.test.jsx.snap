// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StickyMenu component Matches the snapshot 1`] = `
<div
  className="ssb-sticky-menu"
>
  <div
    className="menu-content"
  >
    StickyMenu
  </div>
</div>
`;
